validphys2/src/validphys/config.py

```diff
@@ -1247,6 +1249,8 @@ def parse_default_filter_rules_recorded_spec_(self, spec):
     def parse_added_filter_rules(self, rules: (list, type(None)) = None):
         return rules
 
+    @freeze_args
+    @functools.lru_cache
```

Comment:
I will check in master

Reply:
For NNPDF4.0 with 3 replicas it is called 1001 times. For 2 replicas 693 times.

Reply:
I see. Then indeed, let's keep the freeze args for now, but please add a comment with that info on top. That should definitely not happen since the rules should be the same per replica, so I believe this is a signal of a problem somewhere else which hopefully can eventually be solved.

Reply:
Added comment in 19771a027d0dacfc966c339180680bdeda06954c